api/container_stats.go

```diff
@@ -47,6 +47,12 @@ func (c *API) ContainerStats(ctx context.Context, name string) (Stats, error) {
 		return stats, ContainerNotFound
 	}
 
+	// Since podman 4.6.0, a 200 response with `container is stopped` is returned for stopped containers.
+	var errResponse Error
+	if unmarshalErr := json.Unmarshal(body, &errResponse); unmarshalErr == nil && errResponse.Cause == "container is stopped" {
+		return stats, ContainerNotFound
+	}
```

Comment:
```suggestion
	if _ := json.Unmarshal(body, &errResponse); errResponse.Cause == "container is stopped" {
		return stats, ContainerNotFound
	}
```

I think `unmarshalErr` is irrelevant here? Since `errResponse` is not a pointer and `Cause` is just a string, in case of unmarshelling errors `Cause` would be an empty string.

https://go.dev/play/p/qhoq5l0pYIA

Reply:
> However, I don't generally skip error checking.

That's indeed a very good practice 😄 

I hope this doesn't come back to hunt me, but in this case I think it's OK because we're doing anything with it and...

>  In this case, `errResponse` could be filled with partial data

True, but then we would either have the error message in `Cause` or an empty string right? Or maybe I missing another possible scenario? 😅 

> and in general its value is undefined when err != nil.

Not in this case since `errResponse` is a struct, not a pointer, so we would have all fields be zero values.

Reply:
> True, but then we would either have the error message in Cause or an empty string right? Or maybe I missing another possible scenario? 😅

In practice, I doubt that you are missing anything.

But I think about this with my "unspecified behavior" hat on. If the `Unmarshal` function doesn't say that it will leave the second argument alone if it returns an a non-nil error, I make no assumptions about what it might put there. It could just as well decide that filling in the `"container is stopped"` string in a `Cause` field is something it'll do. And, I couldn't really complain since it returned a non-nil error and didn't say that it will leave the second argument in any particular modified/unmodified state if it does that. I suspect that a failure mid-parse could actually modify the second argument (although, not in a way that I'd expect would change the behavior of this code).

But, I'm being overly pedantic :)